README.md

```diff
@@ -12,6 +12,53 @@
 - Decorrelates [BRIEF](https://www.cs.ubc.ca/~lowe/525/papers/calonder_eccv10.pdf) features under rotational invariance, leading to better performance in nearest-neighbor applications.
 - Unlike [SURF](https://www.vision.ee.ethz.ch/~surf/eccv06.pdf) and [SIFT(Scale-Invariant Feature Transform)](http://weitz.de/sift/), which are patented algorithms, [ORB](http://www.willowgarage.com/sites/default/files/orb_final.pdf) is free to use.
 
+## Overview
+
+The process of generating matches takes two phases; `finding` and `matching`. `finding`, 
+or identifying interest points in an image, is done using the [`findPoints`](/src/orb.core.js#L205) method. It passes a `cornersArray` to the global `utils` object's `getPoints` object property, which can be stored for later use. `finding` will take a few hundred milliseconds for images of standard sizes (~ 720p).
+
+Please note that the the "global `utils` object" mentioned above is returned as a parameter to the callback function from where it can be accessed. See this example:
+
+```js
+  new Matcher('path/to/image1.png', 'path/to/image2.png',
+    async function (r) { // r here is the passed utils object
+      res = await r;
+      console.log(res.getPoints); // see output below
+    });
+```
+The output `getPoints` is in the following format:
+```
+[{"x":37,"y":261},
+ {"x":482,"y":402},
+ {"x":84,"y":331}, ...]
+```
+
+`matching` is done with the [`findMatchedPoints`](/src/orb.core.js#L241) function, which passes a `matchesArray` to the global `utils` object's `getMatchedPoints` object property with the following format:
+```
+[{"confidence":{"c1":63,"c2":187},"x1":359,"y1":48,"x2":65,"y2":309,"population":9},
+ {"confidence":{"c1":124,"c2":169},"x1":260,"y1":333,"x2":546,"y2":295,"population":9}, ...]
+```
+It runs slower than the point `finding` step due to the added computational overhead of comparing both of the images for matches.
+
+The [`findMatchedPoints`](/src/orb.core.js#L241) function depends upon the values served back into its lexical scope by the [`findPoints`](/src/orb.core.js#) function, which in turn depends upon the `params` argument (see below) supplied by the user, and is solely responsible for the generation of the [`cornersArray`](/src/orb.core.js#233), which is used to instantiate the [`matchesArray`](/src/orb.core.js#L269). The [`findMatchesPoints`](/src/orb.core.js#L241), is [called here](/src/orb.core.js#L302) and the appropriate values are set [in the cache](/src/orb.core.js#L316).

```

Comment:
Oops maybe a typo on findmatchespoints here?

Also I really do think a full example here would be ideal in addition to the linked code. It just helps explain each parameter. Thanks!

Reply:
https://github.com/publiclab/matcher-core/blob/c3a5b8a6fccefc5fe7ad521eb40e25a8d9059551/src/orb.core.js#L244

@jywarren The `findMatchedPoints` function doesn't take any values explicity, those are served to it owing to the `findPoints` method. The function, however, depends specifically on `args.browser` that's initialized all the way at the top. Do you want me to highlight this?

https://github.com/publiclab/matcher-core/blob/c3a5b8a6fccefc5fe7ad521eb40e25a8d9059551/src/orb.core.js#L14

Also, every parameter has been explained in the `codeflow` section, along with their line addresses as well.

https://github.com/publiclab/matcher-core/blob/c3a5b8a6fccefc5fe7ad521eb40e25a8d9059551/README.md#codeflow

So do you want me to highlight the fact and maybe link a few lines about the working of `findMatchedPoints` (other than the text above) or modify something else? Sorry for taking your time on this PR, just wanted to be sure we are on the same page here!

Thank you!

Reply:
> The findMatchedPoints function doesn't take any values explicity, those are served to it owing to the findPoints method.

Hm, why doesn't it accept two `points` parameters, between which it finds matches? Wouldn't that make more sense to a user? Like, it would take two points collections as inputs, and output those which are matched between? If this would take extra changes, let's do that in a follow-up PR. Can you open an issue and copy these notes in? Thanks!

> The function, however, depends specifically on args.browser that's initialized all the way at the top. Do you want me to highlight this?

Yes, that doesn't seem so clear to me so it may need explaining. 

So, `Below is a summary of each component of the orbify instance` doesn't clearly spell out that these are parameters that can be submitted in the constructor. I think it's very important to literally show verbatim how to use each of the constructors, as in this example:

![image](https://user-images.githubusercontent.com/24359/61965267-cc127a00-af9d-11e9-8b16-1aef881b067f.png)

That'll make it much easier for people to start using this library, or to use it in unexpected ways! 

Thank you, @rexagod -- much appreciated!